docs/DeveloperGuide.md

```diff
@@ -1,38 +1,361 @@
+
+---
+
+### Before writing code
+
+1. **Configure the coding style**
+   If using IDEA, you can use the following steps to import the code style settings.
+    1. Go to `File → Settings → Editor → Code Style`
+    2. Click the Gear Icon next to the `Scheme` box and then click `Import Scheme → IntelliJ IDEA code style XML`.
+    3. Select the `DefaultCodeStyle.xml` file in the root of the project directory.
+2. **Set up CI**
+   This project comes with a GitHub Actions config files (in `.github/workflows` folder). When GitHub detects those
+   files, it will run the CI for your project automatically at each push to the `master` branch or to any PR. No set up
+   required.
+3. **Learn the design**
+   When you are ready to start coding, we recommend that you get some sense of the overall design by reading
+   about [MoneyMoover’s architecture](DeveloperGuide.md#architecture).
+
+When you are ready to start coding, we recommend that you get some sense of the overall design by reading
+about [MoneyMoover’s architecture](DeveloperGuide.md#architecture).
+
+## Design
+
+### Architecture
+
+![ArchitectureDiagram](images/ArchitectureDiagram.png)
+
+The diagram above provides a high-level overview of how the project is structured. The main components are:
+
+1. The `Main` class which initialises all the other components at startup, and connects them with each other
+2. The `UI` component which is responsible for all user input and output
+3. The `Parser` component which parses user input and creates the relevant Command objects
+4. The `Command` component which executes the logic
+5. The `Accounts` component which manages the user's accounts
+6. The `Forex` (Foreign Exchange) component which handles exchange-rate related logic
+7. The `Transactions` component which manages the user's transactions
+8. The `Storage` component which handles the saving and loading of data to disk
+
+### General Sequence
+
+The following is a high-level sequence of a single `create-account SGD` command, which demonstrates how the components
+interact with each other:
+
+![BasicSequence](images/BasicSequence.png)
+
+### UI component
+
+The API of this component is specified in the `Ui.java`.
+
+![UI Class Diagram](images/UiClassDiagram.png)
+
+The UI class deals with the user interaction with the application, which includes the printing and the logic to read in
+inputs. We will pass in the instance of UI into the `execute` method of the `Command` class. The main features includes:
+
+- Printing text such as new line, spacer, farewell message, greeting message.
+
+- Empowers the users to print customised message in `printf` and `printMessage` method.
+
+- Reads in the user input as `String`.
+
+### Parser component
+
+![Parser Class Diagram](images/ParserClassDiagram.png)
+
+The `Parser` Component
+
+- Parses the user input and creates the relevant `Command` object
+- Makes use of the `CommandType` enum to determine the type of command to create
+
```

Comment:
I like how the guide was broken into components with each block contain greater details